dashboard/app/controllers/admin_users_controller.rb

```diff
@@ -100,4 +100,39 @@ def manual_pass
     flash[:alert] = "UserLevel (ID: #{user_level.id}) updated"
     redirect_to :manual_pass_form
   end
+
+  def permissions_form
+  end
+
+  # Grants the indicated permission to the indicated user. Expects params[:email] and
+  # params[:permission] to be populated.
+  def grant_permission
```

Comment:
Most of that functionality is already there I'd say, via `user.permission = permission`. And the `User` model is bloated enough that I don't think see a gain from adding logic unlikely to be used elsewhere (the logic surrounding `permission` being `admin` or a `UserPermission`).